src/Plugin/KeyType/BasicAuthKeyType.php

```diff
@@ -86,7 +102,7 @@ public function validateKeyValue(array $form, FormStateInterface $form_state, $k
   /**
    * {@inheritdoc}
    */
-  public function getAuthenticationMethod(KeyInterface $key) : Authentication {
+  public function getAuthenticationMethod(KeyInterface $key, KeyInterface $key_token = NULL) : Authentication {
```

Comment:
Key token interface is not needed here, can't you introduce a new optional parameter just in OauthKeyType class's `getAuthenticationMethod()` in the end of the original parameter list (`function getAuthenticationMethod(KeyInterface $key) : Authentication {}`).